pkg/controller/clusters/status.go

```diff
@@ -68,6 +82,16 @@ func UpdateClusterStatusForCNI(ctx context.Context, cluster *anywherev1.Cluster)
 	}
 }
 
+// updateControlPlaneReadyConditionOnExternalETCD updates the ControlPlaneReady condition if etcdadm cluster is not ready.
+func updateControlPlaneReadyConditionOnExternalETCD(cluster *anywherev1.Cluster, kcp *controlplanev1.KubeadmControlPlane, etcdadmCluster *etcdv1.EtcdadmCluster) {
```

Comment:
this name is misleading since it's also called for stacked etcd
```suggestion
func updateConditionsForEtcdAndControlPlane(cluster *anywherev1.Cluster, kcp *controlplanev1.KubeadmControlPlane, etcdadmCluster *etcdv1.EtcdadmCluster) {
```